### PR TITLE
fix: LDIO Ldes Client improvements

### PR DESCRIPTION
**Pull request by Yalz**
# Solves two issues: 

* Unable to restart LDES Client when endpoint was previously unavailable
* Gracefully shutdown LDES Client (wait for pipeline Data IN == Data OUT)

### CODE REVIEW

File: ldi-orchestrator/ldio-application/src/test/java/be/vlaanderen/informatievlaanderen/ldes/ldio/pipeline/PipelineServiceTest.java [hunk 1]
```diff
@@ -2,57 +2,136 @@
...
+	private final PipelineStatusService pipelineStatusService = mock(PipelineStatusServiceImpl.class);
+	private final PipelineFileRepository pipelineRepository = mock(PipelineFileRepository.class);
+	private PipelineServiceImpl pipelineService;
+
+	@BeforeEach
+	void setup() {
+		pipelineService = new PipelineServiceImpl(pipelineCreatorService, pipelineStatusService, pipelineRepository);
+	}
+
+	// TODO: Add tests for the other methods
```
Comment (jobulcke) [inline ldi-orchestrator/ldio-application/src/test/java/be/vlaanderen/informatievlaanderen/ldes/ldio/pipeline/PipelineServiceTest.java:31]: Either implement this TODO or remove please

File: ldi-orchestrator/ldio-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/pipeline/creation/LdioInput.java [hunk 2]
```diff
@@ -67,6 +68,10 @@ protected void processModel(Model model) {
 
 	@SuppressWarnings("java:S6916")
 	public PipelineStatus updateStatus(PipelineStatusTrigger trigger) {
+		return updateStatus(trigger, MANUAL);
```
Comment (jobulcke) [inline ldi-orchestrator/ldio-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/pipeline/creation/LdioInput.java:71]: Good catch ;)

File: ldi-orchestrator/ldio-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/pipeline/exception/PipelineInitialisationException.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package be.vlaanderen.informatievlaanderen.ldes.ldio.pipeline.exception;
+
+public class PipelineInitialisationException extends PipelineException {
+	private final String pipelineName;
+	private final Exception exception;
+
+	public PipelineInitialisationException(String pipelineName, Exception exception) {
```
Comment (jobulcke) [inline ldi-orchestrator/ldio-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/pipeline/exception/PipelineInitialisationException.java:7]: I would rather like to see the abstract super class also implement the `super(Throwable cause)` constructor and pass this exception through as the cause

File: ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java [hunk 4]
```diff
@@ -71,7 +73,13 @@ private synchronized void run() {
 				processModel(memberSupplier.get().getModel());
 			}
 		} catch (EndOfLdesException e) {
+			log.info("SHUTTING DOWN pipeline {} because end of LDES has been reached", pipelineName);
 			shutdownPipeline();
+		} catch (HttpRequestException e) {
+			updateStatus(PipelineStatusTrigger.HALT, StatusChangeSource.AUTO);
+			clientStatusConsumer.accept(ClientStatus.ERROR);
+			log.error("LDES URL unavailable. Client paused: {}", e.getMessage());
+			run();
```
Comment (jobulcke) [inline ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java:82]: Oh, I was not sure anymore if the user should restart the client or pipeline by them self or that the workbench is responsible for this

File: ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java [hunk 5]
```diff
@@ -109,17 +117,15 @@ protected void pause() {
...
-			while (Boolean.FALSE.equals(canGracefullyShutdownChecker.get())) {
-				try {
-					Thread.sleep(Duration.ofSeconds(30));
-				} catch (InterruptedException e) {
-					Thread.currentThread().interrupt();
-				}
+		threadRunning = false;
+		do {
+			try {
+				Thread.sleep(Duration.ofSeconds(1));
```
Comment (jobulcke) [inline ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java:123]: Isn't that very intensive to run this every second?

**Review by jobulcke — APPROVED** [review]

File: ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java (hunk 4 above)
Comment (Yalz) [reply ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java:82]: The use case came, it was easily done..

File: ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java (hunk 5 above)
Comment (Yalz) [reply ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java:123]: Keep in mind, this only happens when shutting down the client or when an exception is reached. 
In the case of a manual shutdown, you want to have your http response as soon as possible. 
Also, the throughput should be quite fast

**Pull request merged**